Jakefile

```diff
@@ -31,6 +31,19 @@ task('minify', function () {
   console.log('Minification completed.');
 });
 
+task('doc', function () {
+  exec('./node_modules/.bin/rimraf out');
+  exec('./node_modules/.bin/jsdoc -c jsdoc.json lib/* docs/jsdoc/*');
+  console.log('Documentation generated.');
+})
+
+task('docPublish', ['doc'], function () {
+  console.log('Pushing docs to gh-pages...');
+  console.log('If this fails, run git checkout -f master to get back to a clean state.');
```

Comment:
If someone runs this without push access to the repo, that is the only way to get back to the master branch as far as I know.